dump.go

```diff
@@ -0,0 +1,97 @@
+package jet
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+)
+
+// dumpAll returns
+//  - everything in Runtime.context
+//  - everything in Runtime.variables
+//  - everything in Runtime.set.globals
+//  - everything in Runtime.blocks
+func dumpAll(a Arguments, depth int) reflect.Value {
+	var b bytes.Buffer
+	var vars VarMap
+
+	ctx := a.runtime.context
+	fmt.Fprintln(&b, "Runtime.context:")
+	fmt.Fprintf(&b, "\t%s %q\n", ctx.Type(), ctx)
+
+	fmt.Fprintln(&b, "Runtime.variables:")
```

Comment:
can we just call `dumpScope(&b, a.runtime, 0)` here? and then wrap that call in a loop that calls dumpScope with each depth until the requested depth is reached?

Reply:
resolved